app/models/generic_object.rb

```diff
@@ -0,0 +1,83 @@
+class GenericObject < ApplicationRecord
+  validates :name, :presence => true, :uniqueness => true
+
+  belongs_to :generic_object_definition
+  has_many   :custom_attributes, :as => :resource, :dependent => :destroy, :autosave => true
+
+  TYPE_MAP = {
+    :boolean  => ActiveRecord::Type::Boolean.new,
+    :datetime => ActiveRecord::Type::DateTime.new,
+    :time     => ActiveRecord::Type::Time.new,
+    :float    => ActiveRecord::Type::Float.new,
+    :integer  => ActiveRecord::Type::Integer.new,
+    :string   => ActiveRecord::Type::String.new,
+  }.freeze
+
+  def initialize(klass, attributes = {})
+    attributes[:generic_object_definition] = definition_object(klass)
+    super(attributes)
+  end
+
+  def inspect
+    attributes_as_string = self.class.column_names.collect do |name|
+      "#{name}: #{attribute_for_inspect(name)}"
+    end
+
+    attributes_as_string += custom_attributes.collect do |ca|
+      value = type_cast(ca.name, ca.value)
+      "#{ca.name}: #{ca_attribute_for_inspect(value)}"
+    end
+
+    "#<#{self.class} #{attributes_as_string.compact.join(", ")}>"
+  end
+
+  def method_missing(method_name, *args)
+    m = method_name.to_s.chomp("=")
+    super unless definition_object.defined_attributes.include?(m)
+
+    method_name.to_s.end_with?('=') ? custom_attribute_setter(m, args.first) : custom_attribute_getter(m)
+  end
+
+  def respond_to_missing?(method_name, *args)
+    raise "Specify the GenericObjectDefinition to create a GenericObject." unless definition_object
+    definition_object.defined_attributes.include?(method_name.to_s.chomp!("=")) || super
+  end
+
+  def property_attributes
+    custom_attributes(true).each_with_object({}) { |ca, h| h[ca.name] = type_cast(ca.name, ca.value) }
+  end
+
+  private
+
+  def definition_object(klass = nil)
+    @definition_object ||= klass ? GenericObjectDefinition.find_by(:name => klass) : generic_object_definition
+  end
+
+  def custom_attribute_getter(name)
+    ca = custom_attributes.find_by(:name => name)
+    ca ? type_cast(ca.name, ca.value) : nil
+  end
+
+  def custom_attribute_setter(name, value)
+    ca = custom_attributes.find_by(:name => name)
+    ca ? ca.update(:value => value.to_s) : custom_attributes.new(:name => name, :value => value.to_s)
```

Comment:
If you are updating one side you are writing to the database, but on the other side you aren't until the caller calls .save.  I think it should be consistent.